FirebaseMLModelDownloader/Tests/Unit/ModelDownloaderTests.swift

```diff
@@ -139,6 +139,50 @@ final class ModelDownloaderTests: XCTestCase {
     XCTAssertEqual(modelInfoRetriever.modelInfo?.size, 562_336)
   }
 
+  /// Test to download model file - makes an actual network call.
+  // TODO: Move this into a separate integration test and add unit test with mocks.
+  func testStartModelDownload() {
+    let testApp = FirebaseApp.app()!
+    let functionName = #function
+    let testModelName = "\(functionName)-test-model"
+    let modelInfoRetriever = ModelInfoRetriever(
+      app: testApp,
+      modelName: testModelName
+    )
+    modelInfoRetriever.modelInfo = ModelInfo(
```

Comment:
I curious why it's needed? Also it doesn't look like 100% safe to modify the object private state. Should we update the class API to protect from unintentional unsafe actions?

Reply:
Typically, the only way that modelInfo is set is after downloading the info from server. I wanted to test the model file download functionality without making another call for model info, so this is a workaround until I separate out the integration and unit tests. Good point though, I will lock down ModelInfo from being externally modifiable in a future PR.